### PR TITLE
Feature datagrid

### PR DESCRIPTION
**Pull request by sccolbert**
This implements a high performance 2D datagrid capable of scaling to over a quadrillion rows. Currently we have a "readonly" feature set. I want to release the `0.1.0` version before implementing things like selections, cell editors, etc. 

This initial version supports full styling, arbitrary cell renders, effectively unlimited size data models, and interactive row/column resizing.

### CODE REVIEW

**Comment by sccolbert** [comment]
CC @blink1073 @afshin @jasongrout @ellisonbg

File: packages/widgets/src/gridlayout.ts [hunk 1]
```diff
@@ -0,0 +1,883 @@
...
+    if (this.parent) {
+      this.parent.fit();
+    }
+  }
+
+  /**
+   * Get the number of columns in the layout.
+   */
+  get columnCount(): number {
+    return this._columnSizers.length
```
Comment (blink1073) [inline packages/widgets/src/gridlayout.ts]: Missing semicolon

File: packages/datagrid/src/datagrid.ts [hunk 2]
```diff
@@ -0,0 +1,3701 @@
...
+  resizeSection(area: 'row' | 'column' | 'row-header' | 'column-header', index: number, size: number): void {
+    this._resizeSection(this._getSectionList(area), index, size);
+  }
+
+  /**
+   * Reset sections in the data grid to their base size.
+   *
+   * @param area - The grid area for the section of interest.
+   */
+  resetSections(area: 'row' | 'column' | 'row-header' | 'column-header'): void {
```
Comment (blink1073) [inline packages/datagrid/src/datagrid.ts]: Should this be `resetSection` for consistency with `resizeSection`?

Comment (sccolbert) [inline packages/datagrid/src/datagrid.ts]: No. Because it's not taking an index. It's resetting *all* sections withing that area.

**Pull request merged**